### PR TITLE
Implement as_observable for event_based_mail

### PR DESCRIPTION
**Pull request by Neverlord**
Relates #1745.

### CODE REVIEW

File: libcaf_core/caf/event_based_mail.test.cpp [hunk 1]
```diff
@@ -243,6 +244,83 @@ TEST("send request message") {
...
+        return {
+          [](int value) { return value * value; },
+        };
+      });
+      auto err = std::make_shared<error>();
+      auto result = std::make_shared<int>(0);
+      SECTION("valid response") {
+        self->mail(3)
+          .request(dummy, infinite)
+          .as_observable<int>()
```
Comment (dominiklohmann) [inline libcaf_core/caf/event_based_mail.test.cpp:259]: Can this infer the type for mail sent to a typed actor?

Comment (Neverlord) [reply libcaf_core/caf/event_based_mail.test.cpp:259]: Sure, the test for that is below (line 306).

Comment (dominiklohmann) [reply libcaf_core/caf/event_based_mail.test.cpp:259]: I missed that, thanks! 🙈

File: libcaf_core/caf/event_based_mail.test.cpp [hunk 2]
```diff
@@ -243,6 +244,83 @@ TEST("send request message") {
       expect<exit_msg>().to(dummy);
     }
   }
+  SECTION("using .to_observable for the response") {
+    SECTION("valid response") {
```
Comment (riemass) [inline libcaf_core/caf/event_based_mail.test.cpp:248]: We have "valid response" nested twice, maybe this description would be a better match with the below section.
```suggestion
    SECTION("response from dynamically typed receiver") {
```

**Review by riemass — CHANGES_REQUESTED** [review]
Small nit, looks good otherwise :+1: .

File: libcaf_core/caf/event_based_mail.hpp [hunk 3]
```diff
@@ -46,7 +46,7 @@ class event_based_scheduled_mail_t
   [[nodiscard]] auto request(const Handle& receiver, timespan relative_timeout,
                              RefTag ref_tag = {},
                              SelfRefTag self_ref_tag = {}) && {
-    detail::send_type_check<typename Trait::signatures, Handle, Args...>();
+    detail::send_type_check<none_t, Handle, Args...>();
```
[the remark this answers is not in the document]
Comment (Neverlord) [reply libcaf_core/caf/event_based_mail.hpp:49]: The signatures here is used to check whether the sender can receive the response in its regular behavior. However, a `request` does not use the regular behavior. Hence, we must skip this check.

**Review by riemass — APPROVED** [review]
Thanks for the explanation.

**Pull request merged**